### PR TITLE
Don't raise exceptions splitting a blank string

### LINKED ISSUE #20002: 0.22.0 - Splitting blank string causes exception
#### Code Sample, a copy-pastable example if possible

```python
import pandas
df = pandas.DataFrame({'test': ['']})
df.test.str.split(expand=True)
```

#### Problem description
Splitting a blank (empty or whitespace) string causes an exception when `expand=True`.

```
IndexError                                Traceback (most recent call last)
<ipython-input-22-a7876ad70d70> in <module>()
      1 import pandas
      2 df = pandas.DataFrame({'test': ['']})
----> 3 df.test.str.split(expand=True)
      4

~/.pyenv/versions/3.6.3/envs/picking/lib/python3.6/site-packages/pandas/core/strings.py in split(self, pat, n, expand)
   1479     def split(self, pat=None, n=-1, expand=False):
   1480         result = str_split(self._data, pat, n=n)
-> 1481         return self._wrap_result(result, expand=expand)
   1482
   1483     @copy(str_rsplit)

~/.pyenv/versions/3.6.3/envs/picking/lib/python3.6/site-packages/pandas/core/strings.py in _wrap_result(self, result, use_codes, name, expand)
   1427                 # propogate nan values to match longest sequence (GH 18450)
   1428                 max_len = max(len(x) for x in result)
-> 1429                 result = [x * max_len if x[0] is np.nan else x for x in result]
   1430
   1431         if not isinstance(expand, bool):

~/.pyenv/versions/3.6.3/envs/picking/lib/python3.6/site-packages/pandas/core/strings.py in <listcomp>(.0)
   1427                 # propogate nan values to match longest sequence (GH 18450)
   1428                 max_len = max(len(x) for x in result)
-> 1429                 result = [x * max_len if x[0] is np.nan else x for x in result]
   1430
   1431         if not isinstance(expand, bool):

IndexError: list index out of range
```

#### Expected Output
Either nans or empty strings as placeholder values in the output dataframe.

```
Out[24]:
   0
0  ''
```

#### Output of ``pd.show_versions()``

<details>

[paste the output of ``pd.show_versions()`` here below this line]
INSTALLED VERSIONS
------------------
commit: None
python: 3.6.3.final.0
python-bits: 64
OS: Darwin
OS-release: 17.4.0
machine: x86_64
processor: i386
byteorder: little
LC_ALL: None
LANG: en_US.UTF-8
LOCALE: en_US.UTF-8

pandas: 0.22.0
pytest: None
pip: 9.0.1
setuptools: 28.8.0
Cython: None
numpy: 1.14.0
scipy: 0.19.1
pyarrow: None
xarray: None
IPython: 6.1.0
sphinx: None
patsy: 0.4.1
dateutil: 2.6.1
pytz: 2017.2
blosc: None
bottleneck: None
tables: 3.4.2
numexpr: 2.6.2
feather: None
matplotlib: 2.0.2
openpyxl: None
xlrd: None
xlwt: None
xlsxwriter: None
lxml: None
bs4: None
html5lib: 0.9999999
sqlalchemy: 1.1.14
pymysql: None
psycopg2: 2.7.3.1 (dt dec pq3 ext lo64)
jinja2: 2.9.6
s3fs: None
fastparquet: None
pandas_gbq: None
pandas_datareader: None
</details>

### PR DESCRIPTION
**Pull request by montanalow**
whatsnew: bug fix on reshaping
- [X] closes #20002
- [x] tests added / passed
- [X] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`

### CODE REVIEW

File: pandas/tests/test_strings.py [hunk 1]
```diff
@@ -1943,6 +1943,13 @@ def test_split(self):
                       [u('f'), u('g'), u('h')]])
         tm.assert_series_equal(result, exp)
 
+        # expand blank split
```
Comment (jreback) [inline pandas/tests/test_strings.py]: can you make a separate test

File: pandas/tests/test_strings.py [hunk 2]
```diff
@@ -1943,6 +1943,13 @@ def test_split(self):
...
+        values = Series(['a b c', 'a b', '', ' '])
```
Comment (jreback) [inline pandas/tests/test_strings.py]: can you also test on a single string, like the original issue. and add the issue number as a comment.

**Comment by jreback** [comment]
this looks ok, though failing some other tests. you may need to adjust them.

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/tests/test_strings.py [hunk 3]
```diff
@@ -1992,6 +1992,19 @@ def test_rsplit(self):
         exp = Series([['a_b', 'c'], ['c_d', 'e'], NA, ['f_g', 'h']])
         tm.assert_series_equal(result, exp)
 
+    def test_split_blank_string(self):
+        # expand blank split GH 20067
+        values = Series([''])
```
Comment (jreback) [inline pandas/tests/test_strings.py]: can you add a ``name=`` to the Series so its clear where the name goes.

**Review by jreback — CHANGES_REQUESTED** [review]
can you add a whatsnew note. bug fix on reshaping is ok.

File: pandas/tests/test_strings.py (hunk 3 above)
Comment (montanalow) [reply pandas/tests/test_strings.py]: Not sure I follow exactly what you mean. Are you suggesting adding `name='test'` as an arg to the constructor?

Comment (jreback) [reply pandas/tests/test_strings.py]: yes add a name, the point is that this generates the column name for the frame.

**Review by jreback — CHANGES_REQUESTED** [review]

**Comment by montanalow** [comment]
@jreback It looks like travis and appveyor canceled their builds, after I added names per request. Is there a retry mechanism for them?

**Comment by jreback** [comment]
looks fine. can you rebase on master & add a whatsnew note? ping on green.

**Comment by montanalow** [comment]
@jreback all green

**Comment by TomAugspurger** [comment]
@montanalow can you add a release note to whatsnew/v0.23.0.txt under bug fixes? LGTM otherwise.

**Comment by montanalow** [comment]
@TomAugspurger note added

**Comment by TomAugspurger** [comment]
Thanks @montanalow !

**Pull request merged**